array.c

```diff
@@ -3097,20 +3097,22 @@ rb_ary_to_s(VALUE ary)
  *  call-seq:
  *    to_a -> self or new_array
  *
- *  Returns +self+ if <tt>self.instance_of?(Array)</tt>:
+ *  When <tt>self.instance_of?(Array)</tt>, returns +self+:
  *    a = [:foo, 'bar', 2]
+ *    a.instance_of?(Array) # => true
  *    a1 = a.to_a
- *    a1 # => [:foo, "bar", 2]
  *    a1.equal?(a) # => true # Returned self
  *
- *  If +self+ is a subclass of \Array, returns the new \Array
- *  formed by converting +self+ to an \Array:
+ *  Otherwise, when <tt>self.kind_of?(Array)</tt>,
+ *  returns a new \Array containing the actual elements (not copies) of +self+:
```

Comment:
`Otherwise, returns an instance of \Array containing the elements of +self+`.  I don't see the value in making a distinction for actual elements vs. copies.

Reply:
Thanks, @jeremyevans.  It's useful to know that elements are generally not copied.  (I have only studied the methods I've written about so far.)  I had not thought anything about copies until I ran into the copied hash key.
I'll get this cleaned up.  Again, thanks for your careful and helpful reviewing.